interfaces/builtin/fwupd.go

```diff
@@ -64,13 +68,78 @@ const fwupdPermanentSlotAppArmor = `
   /sys/firmware/efi/esrt/entries/** r,
 
   # Allow fwupd to access system information
-  /sys/devices/virtual/dmi/id/product_name r,
-  /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/smbios_entry_point r,
+  /sys/devices/virtual/dmi/id/{sys,bios,product,chassis,board}_{name,version,vendor,date,serial,sku,type,asset_tag,family} r,
+
+  /sys/firmware/dmi/tables/DMI r,
+  /sys/bus/usb/devices/ r,
+
+  /sys/firmware/acpi/bgrt/{type,version,{x,y}offset,image} r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
+  /proc/modules r,
+  /run/udev/data/* r,
+  /sys/devices/**/usb*/busnum r,
+  /sys/devices/**/usb*/**/busnum r,
+  /sys/devices/**/usb*/devnum r,
+  /sys/devices/**/usb*/**/devnum r,
+  /sys/devices/**/usb*/speed r,
+  /sys/devices/**/usb*/**/speed r,
+  /sys/devices/**/usb*/descriptors r,
+  /sys/devices/**/usb*/**/descriptors r,
+  /sys/devices/**/usb*/bConfigurationValue r,
+  /sys/devices/**/usb*/**/bConfigurationValue r,
+  /dev/mem r,
+  /sys/kernel/security/lockdown r,
+  owner @{PROC}/@{pid}/mountinfo r,
+  /sys/bus/pci/devices/ r,
+  /sys/devices/pci*/**/vendor r,
+  /sys/devices/pci*/**/device r,
+  /sys/devices/pci*/**/revision r,
+  /sys/devices/pci*/**/subsystem_vendor r,
+  /sys/devices/pci*/**/subsystem_device r,
+  /sys/devices/pci*/**/class r,
+  /sys/devices/pci*/**/config r,
```

Comment:
Need `/sys/class/pci_bus/*/device/config` for the PCI BIOS control register plugin

Reply:
The canonical path should match `/sys/devices/pci*/**/config`.